arch/arm/soc/st_stm32/stm32f2/Kconfig.defconfig.series

```diff
@@ -12,4 +12,14 @@ gsource "arch/arm/soc/st_stm32/stm32f2/Kconfig.defconfig.stm32f2*"
 config SOC_SERIES
 	default "stm32f2"
 
+if GPIO_STM32
+
+config GPIO_STM32_PORTD
```

Comment:
The attached CubeMX project sample is based on the smallest stm32f205rb SOC in stm32f2 series.
In this project, beside the default GPIO A, B, C, the D, H ports are used.
(Pin5 - GPIOH0, Pin6 – GPIOH1, Pin54 – GPIOD2).
Probably this configuration is reflected in this commit.
I can imagine another project/board based on the same SOC with Pin 5,6 and 54 dedicated to OSC_IN, OSC_OUT and UART5_RX respective. In this case configuring the GPIOD and GPIOH is no more necessary.
As I see, at this point we can’t presume the concrete board GPIO configuration.
Moreover, in an extreme situation, some of the default stm32 GPIO A, B, C configuration settings are no more necessary, taking into consideration that the respective pins are occupied by the alternate functions.
[stm32f205rb.zip](https://github.com/zephyrproject-rtos/zephyr/files/2112003/stm32f205rb.zip)

Reply:
@ibisz, from what I see in stm32f2**xx.h files, Ports A to I are all supported in all stm32f2 SoCs.
If this is actually the case (tbc) matching Kconfig symbols should be made available/visible to all STM32F2 SoCs (hence in this file).
Technically, this is redundant with what is done in stm32f2.dtsi file (hence should be improved in future), and then both Kconfig symbols and dts files should be in synch.

So as you say, we should not presume what configuration will be required for a board based on one of these SoCs so all symbols should be available (if this is confirmed matching peripherals are actually available on these SoCs)

Reply:
I select a chip with minimum pin(A, B, C, D, H) in stm32f2 series, So Just leave those gpio port to Kconfig.defconfig.series.

Fewer ports can reduce memory usage although just few bytes. Others SOC can enable other GPIO ports in it's own config.

Reply:
@zhaozhongchen , I got your point, but from experience, we prefer to get all GPIO enabled and activated by default at SoC/board level. Then, they can be disabled at application level for optimisation (or at board level for some boards that have really small footprint, 8K of RAM for instance).
There are 2 reasons:
-This is source of error: You don't get any error if you try to use one pin of a deactivated GPIO. So, this is a usually stupid but long issue to debug, specially for a new comer. 
-We prefer to provide stable configurations. If you don't activate it now, someone will activate it later (here at SoC level) because he needs it. Then your configuration will change and you won't be informed. If you need it to be deactivated in you application, enable it here, and deactivate it in your application, you won't get any surprise later.